### PR TITLE
added deprecation and error messages for 'odo update`

### LINKED ISSUE #3396: deprecation warning for odo update
## Background:
The original intention for  `odo update`  was to provide a path forward for outer-loop by switching to git component.
We don't need that anymore as there is work in progress for proper outer-loop (`odo deploy` `odo pipelines`)

## Acceptance criteria 
- [ ] When `odo update` command is executed in a directory with devfile component it should display error: 
  ```
  'odo update' command is not available for Devfile based components.
  ```
- [ ] When `odo update` is executed with LocalConfig component it should display the following warning on as the first line of the output: 
  ```
  WARNING: 'odo update' command will be removed in the future odo version.
  You should be using `odo config` command instead.
  example:
    odo config set SourceType git
    odo config set SourceLocation https://github.com/example/example
  ```

/kind user-story
/priority high
/area update

### PR DESCRIPTION
**Pull request by dev-gaur**
/kind deprecation

**What does does this PR do / why we need it**:
added deprecation and error messages for `odo update`

**Which issue(s) this PR fixes**:
Fixes #3396 

**How to test changes / Special notes to the reviewer**:
run `odo update` for 1. s2i components (expect a deprecation warning)
 2. devfile component (expect an error)

Its a Draft right now

### CODE REVIEW

**Comment by girishramnani** [comment]
adding WIP in the label as this is still draft PR

**Comment by girishramnani** [comment]
@dev-gaur please make this as non-draft PR by clicking the `Ready for review` button

File: pkg/odo/cli/component/update.go [hunk 1]
```diff
@@ -44,6 +49,14 @@ var updateCmdExample = ktemplates.Examples(`  # Change the source code path of c
 	  %[1]s --binary ./downloads/sample.war
 		`)
 
+const deprecationWarning string = "WARNING: 'odo update' command will be removed in the future odo version.\n" +
+	"You should be using `odo config` command instead.\n" +
```
Comment (adisky) [inline pkg/odo/cli/component/update.go:53]: I think we should add the same warning  in `odo update --help` 
Also can we change the color of message to yellow?

**Comment by girishramnani** [comment]
Please address @adisky , everything else good.
/approve

File: pkg/odo/cli/component/update.go (hunk 1 above)
Comment (dev-gaur) [reply pkg/odo/cli/component/update.go]: image


changes made.

**Pull request merged**